### PR TITLE
224A: optimize coding-agent workflow and harden env/profile test contracts

### PR DESCRIPTION
**Pull request by mpieniak01**
## Summary
This PR delivers the 224/224A agent-workflow optimization track and includes final hardening fixes from regression runs:

- standardizes coding-agent operating contract (brief-first, bounded scope, patch-first flow)
- introduces/extends preflight context and lane-oriented execution guidance
- improves test-lane and gate ergonomics for coding agents
- hardens environment profile contract tests for mixed runtime setups
- fixes CI instability where optional `torch` was treated as a hard dependency in a probe regression test

### Agent workflow optimization (224A)
- prompt/template/process updates to keep agent context small and explicit
- CI/full-CI knowledge/summary integration aligned with lightweight preflight briefing
- lane-aware workflow improvements for predictable target testing

### Regression hardening added in this push
- `tests/test_env_contracts.py`
  - `ollama` profile contract now accepts both:
    - direct `http://localhost:11434...`
    - gateway path via `multi_runtime` (`http://localhost:8014...`)
  - removes false negatives for valid operator setups.
- `tests/test_multi_runtime_probe_logits_tokens.py`
  - converts `torch` dependency to optional (`pytest.importorskip("torch")` inside test)
  - CI environments without `torch` now skip this test instead of failing during collection.

## Why
- We had a branch-vs-main inconsistency symptom in CI caused by optional dependency assumptions and strict env-profile assertion.
- These fixes make test contracts robust across supported runtime topologies and runner footprints.

## Validation
Executed locally on this branch:

- `pytest -q`  
  - result: `5990 passed, 79 skipped, 1 warning`
- `make pr-fast`  
  - result: `PASS` (all gates green after fix)
- targeted:
  - `pytest -q tests/test_env_contracts.py` → `3 passed, 2 skipped`
  - `pytest -q tests/test_multi_runtime_probe_logits_tokens.py` → `1 passed`

## Notes
- Existing runtime warning about `warmup_audio_engine_if_enabled` not awaited remains a separate concern (not introduced by this change).

## Release Notes

* **New Features**
  * Added context brief generation for coding agents to streamline task execution
  * Added environmental contract testing for multiple runtime profiles (Ollama, vLLM, multi-runtime)
  
* **Documentation**
  * Enhanced agent guidelines with scoped execution procedures and context management best practices
  
* **Chores**
  * Expanded CI/CD workflows to generate and upload agent context artifacts
  * Added new build targets for environment contract validation and agent preparation

[Review Change Stack](https://app.coderabbit.ai/change-stack/mpieniak01/Venom/pull/496?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by mpieniak01** [comment]
Addressed actionable review feedback from Gemini + CodeRabbit in commit 74fe1b26.\n\nImplemented:\n- scripts/agent_hard_gate.sh\n  - removed premature  in changed-files scope loop so flags accumulate across all files\n  - expanded backend scope detection to include root-level config/dependency files (, , , )\n- scripts/generate_agent_context_brief.py\n  - expanded  backend detection for the same root-level config/dependency files\n- tests/test_env_contracts.py\n  - added normalized runtime profile comparison ( == ) for , active runtime, and parametrized profile checks\n- Makefile\n  - fixed  continuation for  /  list\n\nValidation run:\n-  -> \n-  -> syntax OK\n-  -> exit 0\n

**Comment by mpieniak01** [comment]
Addressed actionable review feedback from Gemini + CodeRabbit in commit `74fe1b26`.

Implemented:
- `scripts/agent_hard_gate.sh`
  - removed premature `break` in changed-files scope loop so flags accumulate across all files
  - expanded backend scope detection to include root-level config/dependency files (`pyproject.toml`, `requirements*.txt`, `pytest.ini`, `.pre-commit-config.yaml`)
- `scripts/generate_agent_context_brief.py`
  - expanded `_infer_scope()` backend detection for the same root-level config/dependency files
- `tests/test_env_contracts.py`
  - added normalized runtime profile comparison (`-` == `_`) for `ENV_CONTRACT_PROFILE`, active runtime, and parametrized profile checks
- `Makefile`
  - fixed `.PHONY` continuation for `test-202c7-signed-activation-success` / `test-env-contracts-*` list

Validation run:
- `pytest -q tests/test_env_contracts.py tests/test_make_targets_audit.py` -> `6 passed, 2 skipped`
- `bash -n scripts/agent_hard_gate.sh` -> syntax OK
- `python scripts/generate_agent_context_brief.py --mode preflight --repo-root . --diff-base origin/main` -> exit 0
